resources/src/commonMain/kotlin/ThemedColorProvider.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+@file:JvmName("CommonThemedColorProvider")
+package com.splendo.kaluga.resources
+
+import kotlin.jvm.JvmName
+import kotlin.reflect.KProperty
+
+/** Multiplatform Color Provider that is aware of a Dark and Light mode */
+abstract class ThemedColorProvider {
+
+    /** light version of this color */
+    abstract val lightColor: Color
+
+    /** dark version of this color */
+    abstract val darkColor: Color
```

Comment:
I would rename these 2 properties just `light` and `dark`, is already known they refer to color. What do you think?

Reply:
thought about it, but light and dark are also the name of the params which are strings.
I preferred not to rename those to lightHexValue or ligthValue, but to rename these instead.
Because I prefer to call it with `val somename by themeColor(light = "#fff", dark = "#000")`: very simple
Also, the third and more important variable is called `color`, so `darkColor` and `lightColor` seem consistent to me.

Good point, though